src/data/available-models.ts

```diff
@@ -0,0 +1,63 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+
+// This is a list of all the models that are available to use for the OpenAI API key.
+const availableModels = [
+  "davinci",
```

Comment:
Move this to the database as a migration.
Add the following column s to define their use case

- Available_for_text_chat
- Available_for_image_gen

Reply:
Add additional columns to define the functionality/usecase of the model.
Also add a description column to explain the model as well as a link to the documentation of the model.

Reply:
I created model for storing openAiModels
`model OpenAIModels {
    name                        String    @id
    availableForChat            Boolean
    availableForImageGeneration Boolean
    availableForEmbeddings      Boolean
    availableForAudioTranscript Boolean
    availableForFineTuningJobs  Boolean
    availableForFineTunes       Boolean
    description                 String?
    maxTokens                   Int?
    TrainingDataDate            DateTime?
    deprecated                  Boolean?
    User                        User?     @relation(fields: [userId], references: [id])
    userId                      String?
}`

Created seed file with 5 models for chat: 
`  const gpt4 = await prisma.openAIModels.upsert({
    where: {
      name: "gpt-4",
    },
    update: {},
    create: {
      name: "gpt-4",
      availableForChat: true,
      availableForImageGeneration: false,
      availableForEmbeddings: false,
      availableForAudioTranscript: false,
      availableForFineTuningJobs: false,
      availableForFineTunes: false,
      description:
        "More capable than any GPT-3.5 model, able to do more complex tasks, and optimized for chat. Will be updated with our latest model iteration 2 weeks after it is released.",
      maxTokens: 8192,
      TrainingDataDate: new Date("2021-09-01"),
    },
  });`